### ISSUE TITLE
Question: Should tokens and interactions be delete from the DB on logout?

### ISSUE THREAD

**Issue by SokratisVidros** [panva/node-oidc-provider#368]
Supposedly a user has authenticated once with OIDC. In the DB one token and two session entries will be recorded. Should all the previous records be delete when visiting the logout endpoint? If so, is it something that the db adapter should handle?

Thanks in advance.

**Comment by panva** [comment]
Hi @SokratisVidros, can you be more concrete?

**Comment by SokratisVidros** [comment]
Let me rephrase. Is it suggested to remove all issued tokens and session records for a particular user upon logout? 

I mostly wonder about the best and most secure practices on the topic. Should tokens be kept in the database even long after a user has logged out from the OIDC server or after their expiration?

Hope that helps.

**Comment by panva** [comment]
This is a policy you have to define, I am not in a position to make "best practice" recommendations on the topic. Your OP = Your security profile = Your call on the topic.

**Comment by SokratisVidros** [comment]
Cool thanks
